src/beyondllm/llms/cohere.py

```diff
@@ -0,0 +1,73 @@
+from beyondllm.llms.base import BaseLLMModel, ModelConfig
+from typing import Any, Dict
+from dataclasses import dataclass, field
+import os
+import cohere
+
+@dataclass
+class CohereModel:
+    """
+    Class representing a Language Model (LLM) model using Cohere.
+
+    Example:
+    ```
+    >>> llm = CohereModel(api_key="<your_api_key>", model_kwargs={"temperature": 0.5})
+    ```
+    or
+    ```
+    >>> import os
+    >>> os.environ['COHERE_API_KEY'] = "***********" #replace with your key
+    >>> llm = CohereModel()
+    ```
+    """
+    api_key: str =" "
+    model_kwargs: dict = field(default_factory=lambda: {
+        "temperature": 0.5,
+        "top_p": 1,
+        "max_tokens": 2048,
+    })
```

Comment:
define model_name as the class variable. So that one can define any LLM model from Cohere.